src/rpc/net.cpp

```diff
@@ -300,6 +308,9 @@ static RPCHelpMan getpeerinfo()
         obj.pushKV("session_id", stats.m_session_id);
 
         ret.push_back(std::move(obj));
+        if (opt_peer_id && stats.nodeid == *opt_peer_id) {
```

Comment:
Only need to check `opt_peer_id` here, since we'd never get here if it's set to something different

Reply:
?